src/compiler/parser.ts

```diff
@@ -9063,7 +9063,7 @@ namespace ts {
         if (namedArgRegExCache.has(name)) {
             return namedArgRegExCache.get(name)!;
         }
-        const result = new RegExp(`(\\s${name}\\s*=\\s*)('|")(.+?)\\2`, "im");
+        const result = new RegExp(`(\\s${name}\\s*=\\s*)(('([^']*)')|("([^"]*)"))`, "im");

```

Comment:
Consider using non-capturing groups so that the matching substrings don't have to be pulled out and stored.

Reply:
The capture groups are here because we use them.

Reply:
Oh, you mean the outer captures... Hm, yeah, I guess they can be noncapturing.